https://issues.redhat.com/browse/ACM-20220 Access apiserver token doc (#7951)
open-cluster-management/rhacm-docs: clusters/cluster_lifecycle/cluster_proxy_addon_use.adoc

[#use-cluster-proxy-addon]
= Using cluster proxy add-ons

> Use cluster proxy add-ons to access the managed cluster kube API server or managed cluster services by using different methods. See the following topics to learn more about the different ways you can use cluster proxy add-ons.

Reviewer: Please add a link to the topics. I understand that these topics are in the rest of the section but I'm expecting to see the relevant topics listed right after this intro sentence ("See the following topics..."

Author: Updated

*Required access:* Editor

[#use-cluster-proxy-addon-prereq]
== Prerequisites

* You must xref:../cluster_proxy_addon_config#config-cluster-proxy-addon[enable cluster proxy add-ons].

[#access-kube-apiserver-managed-token]
== Accessing the managed cluster `kube-apiserver` by using a managed cluster user token

Use the cluster proxy add-on to list pods in the default namespace of a managed cluster by completing the following steps:

. Export your cluster proxy URL by running the following command. Replace values where needed:

+
[source,bash]
----
export CLUSTER_PROXY_URL=https://$(oc get route -n multicluster-engine cluster-proxy-addon-user -o=jsonpath='{.spec.host}')/<managed-cluster-name>
----

. Get the `ConfigMap` by running the following command:

+
[source,bash]
----
oc get configmap kube-root-ca.crt -o=jsonpath='{.data.ca\.crt}' > hub-ca.crt
----

> . Transfer your CA certificate by running the following command:

Reviewer: First instance should be expanded:
```suggestion
. Transfer your certificate authority (CA) certificate by running the following command:
```

+
[source,bash]
----
curl --cacert hub-ca.crt -H "Authorization: Bearer $TOKEN" https://$CLUSTER_PROXY_URL/api/v1/namespaces/default/pods 
----
+
*Note:* You need a user token that has permission to list pods in the default namespace of the managed cluster.

[#access-kube-apiserver-hub-token]
== Accessing the `kube-apiserver` of a managed cluster by using a hub cluster user token

The target hub cluster and target managed cluster both need to use the same external identity provider so that both clusters can recognize the same user.

> To list pods in the default namespace by using the the hub cluster `developer ` user token, you must first create the correct `Role` and `RoleBinding` for the `developer` on the managed cluster.

Reviewer: ```suggestion
To list pods in the default namespace by using the the hub cluster `developer` user token, you must first create the correct `Role` and `RoleBinding` for the `developer` on the managed cluster.
```

> If you are using {rhacm}, you can use the `ClusterPermission` API to create `Roles` and RoleBindings from the hub cluster.

Reviewer: ```suggestion
If you are using {rhacm}, you can use the `ClusterPermission` API to create `Role` and `RoleBinding` objects from the hub cluster.
```

See the following example that binds the `get` and `list` permissions for pods to the user `developer`:

> [source,yaml]

Reviewer: Don't you need a `+` here?

Author: It's only needed when it comes after a step (`.`)
